util/src/main/java/io/kubernetes/client/informer/cache/ReflectorRunnable.java

```diff
@@ -223,10 +223,15 @@ private void watchHandler(Watchable<ApiType> watch) {
         continue;
       }
       if (eventType.get() == EventType.ERROR) {
-        String errorMessage =
-            String.format("got ERROR event and its status: %s", item.status.toString());
-        log.error(errorMessage);
-        throw new RuntimeException(errorMessage);
+        if (item.status != null && item.status.getCode() == HttpURLConnection.HTTP_GONE) {
+          log.info("Watch connection expired: {}", item.status.getMessage());
```

Comment:
Should `isLastSyncResourceVersionUnavailable` be set to `true` here? @yue9944882

Related PR: https://github.com/kubernetes-client/java/pull/1543/files#diff-31beacc81d066c2ab68de67a3587fee7bd7ef10d9ba42dc0a180b621751c6bffR160

Reply:
This is a bug which has broken us.
Setting `isLastSyncResourceVersionUnavailable` to `true` will not fix the problem, because `isLastSyncResourceVersionUnavailable` is not referenced in the `while(true)` loop anyway.
We should keep the `throw new RuntimeException`.